Plutarch/Prelude.hs

```diff
@@ -13,17 +14,143 @@ module Plutarch.Prelude (
   (#),
   plet,
   pinl,
-  pcon,
-  pmatch,
   pto,
   pfix,
   Type,
   S,
   PType,
+  PlutusType,
+  PCon (pcon),
+  PMatch (pmatch),
+
+  -- * Integers and integer utilities
+  PInteger,
+  PIntegral (pdiv, pmod, pquot, prem),
+
+  -- * Rational numbers and utilities
+  PRational,
+  pnumerator,
+  pdenominator,
+  pfromInteger,
+  pround,
+
+  -- * Booleans and boolean functions
+  PBool (..),
+  PEq ((#==)),
+  POrd ((#<=), (#<)),
+  pif,
+  pnot,
+  (#&&),
+  (#||),
+
+  -- * Bytestrings and bytestring utilities
+  PByteString,
+  phexByteStr,
+  pconsBS,
+  psliceBS,
+  plengthBS,
+  pindexBS,
+
+  -- * String and string utilities
+  PString,
+  pencodeUtf8,
+  pdecodeUtf8,
+
+  -- * Unit type and utilities
+  PUnit (..),
+
+  -- * Common list typeclass and utilities
+  PListLike (PElemConstraint, pelimList, pcons, pnil, phead, ptail, pnull),
+  PIsListLike,
+  plistEquals,
+  pelem,
+  plength,
+  ptryIndex,
+  pdrop,
+  psingleton,
+  pconcat,
+  pzipWith,
+  pzipWith',
+  pzip,
+  pmap,
+  pfilter,
+  precList,
+  pfoldr,
+  pfoldrLazy,
+  pfoldl,
+  pall,
+  pany,
+
+  -- * Scott encoded list type
+  PList (..),
+
+  -- * Scott encoded maybe type and utilities
+  PMaybe (..),
```

Comment:
These should probably be renamed before being added to prelude. Otherwise importing the ledger api becomes a bit painful.

Reply:
Well, maybe. But is that the compromise we're settling on regarding this scott-encoding/data (/builtin) encoding ambiguity issue?